avm/ptn/sa/modernize-your-code/main.bicep

```diff
@@ -0,0 +1,549 @@
+    logAnalyticsWorkspaceResourceId: enableMonitoring ? logAnalyticsWorkspace.outputs.resourceId : ''
+    zoneRedundant: enableRedundancy
+    secondaryLocation: enableRedundancy && !empty(secondaryLocation) ? secondaryLocation : ''
+    privateNetworking: enablePrivateNetworking
+      ? {
+          virtualNetworkResourceId: network.outputs.vnetResourceId
+          subnetResourceId: network.outputs.subnetPrivateEndpointsResourceId
+        }
+      : null
+    tags: allTags
+    enableTelemetry: enableTelemetry
+  }
+}
+
+var containerAppsEnvironmentName = 'cae-${resourcesName}'
+
+module containerAppsEnvironment 'br/public:avm/res/app/managed-environment:0.11.2' = {
+  name: take('container-env-${resourcesName}-deployment', 64)
+  #disable-next-line no-unnecessary-dependson
+  dependsOn: [applicationInsights, logAnalyticsWorkspace, network] // required due to optional flags that could change dependency
+  params: {
+    name: containerAppsEnvironmentName
+    infrastructureResourceGroupName: '${resourceGroup().name}-ME-${containerAppsEnvironmentName}'
+    location: location
+    zoneRedundant: enableRedundancy && enablePrivateNetworking
+    publicNetworkAccess: 'Enabled' // public access required for frontend
+    infrastructureSubnetResourceId: enablePrivateNetworking ? network.outputs.subnetWebResourceId : null
+    managedIdentities: {
+      userAssignedResourceIds: [
+        appIdentity.outputs.resourceId
+      ]
+    }
+    appInsightsConnectionString: enableMonitoring ? applicationInsights.outputs.connectionString : null
+    appLogsConfiguration: enableMonitoring
+      ? {
+          destination: 'log-analytics'
+          logAnalyticsConfiguration: {
+            customerId: logAnalyticsWorkspace.outputs.logAnalyticsWorkspaceId
+            sharedKey: logAnalyticsWorkspace.outputs.primarySharedKey
+          }
+        }
+      : {}
+    workloadProfiles: enableRedundancy
+      ? [
+          {
+            maximumCount: 3
+            minimumCount: 3
+            name: 'CAW01'
+            workloadProfileType: 'D4'
+          }
+        ]
+      : [
+          {
+            name: 'Consumption'
+            workloadProfileType: 'Consumption'
+          }
+        ]
+    tags: allTags
+    enableTelemetry: enableTelemetry
+  }
+}
+
+module containerAppFrontend 'br/public:avm/res/app/container-app:0.17.0' = {
+  name: take('container-app-frontend-${resourcesName}-deployment', 64)
+  params: {
+    name: take('ca-${resourcesName}frontend', 32)
+    location: location
+    environmentResourceId: containerAppsEnvironment.outputs.resourceId
+    managedIdentities: {
+      userAssignedResourceIds: [
+        appIdentity.outputs.resourceId
+      ]
+    }
+    containers: [
+      {
+        env: [
+          {
+            name: 'API_URL'
+            value: 'https://${containerAppBackend.outputs.fqdn}'
+          }
+        ]
+        image: 'cmsacontainerreg.azurecr.io/cmsafrontend:latest'
```

Comment:
hardcoding latest image is not a good idea, specially if at some point, there is a non backward compatible change that depends on the infrastructure (e.g. depend of a new environment variable).

Recommended to create a parameter with a default value set to a specific version of the image that works for the current - e.g. `param frontendContainerImageTag string = 'latest_2025-06-17_704'`